### PR TITLE
Carbon Mathigon Dropdown Style

### PR DESCRIPTION
**Pull request by vabarbosa**
continuation of https://github.com/qiskit-community/platypus/pull/216

preview URL: https://platypus-pr-218.ecy0akwlcpw.us-south.codeengine.appdomain.cloud/course/ch-states/the-atoms-of-computation

---

This PR overrides the mathigon style of the multi-choice dropdowns, in order to align with Carbon's design principles.

Wrong and right answers:
image

Dropdown:
image

Fixes [#2092](https://github.com/Qiskit/qiskit.org/issues/2092).

### CODE REVIEW

**Comment by vabarbosa** [comment]
@muHashh please continue the work from https://github.com/qiskit-community/platypus/pull/216 in this PR.
this PR allows for the preview URLs.

FYI @techtolentino

**Comment by vabarbosa** [comment]
comment from Russell:

> Regarding the red color, it should be red-50, which will make it accessibility compliant. I'll update the design file.
> 
> Regarding the 24px issue, can you share a preview link so I can also see it on a live page?

**Comment by vabarbosa** [comment]
https://github.com/qiskit-community/platypus/pull/216#issuecomment-982903618
> Regarding the red color, it should be red-50, which will make it accessibility compliant. I'll update the design file.
> Regarding the 24px issue, can you share a preview link so I can also see it on a live page?

@JRussellHuffman you can find the fill in the blanks and preview the updates here: 
- https://platypus-pr-218.ecy0akwlcpw.us-south.codeengine.appdomain.cloud/course/ch-states/the-atoms-of-computation

**Comment by JRussellHuffman** [comment]
Thanks for the preview link.
The main issue is see is the gray on gray background, which makes it look less clickable. In the design component file, there are variations for when the background is gray vs white. See screenshot. 
Screen Shot 2021-12-02 at 9 49 38 AM

**Comment by JRussellHuffman** [comment]
Regarding the 24px touch target size, it worked okay on mobile for me, but that is a bit small. But any bigger, would it change the line height for the text around it?

What is the size of the default Mathigon component this is based off of?

**Comment by vabarbosa** [comment]
> What is the size of the default Mathigon component this is based off of?

@JRussellHuffman the Mathigon [`x-blank`](https://github.com/mathigon/studio/blob/main/frontend/components/blank) component:

https://mathigon.org/course/euclidean-geometry/construction
image

appears to have a `font-size: 18px` and `line-height: 24px` for the inline blank (i.e., `???`) and `font-size: 16px` and `line-height: 22px` for the dropdown choices

**Comment by JRussellHuffman** [comment]
Good to know, thanks. Our component is designed to be 24px height and 14pt font for both the inline blank and dropdown choices. I think that will be better. I don't think we should go below 24px for the target area, and the font size is our standard. In which case the only implementation issue I see is the background color. Does all that sound good?

**Comment by vabarbosa** [comment]
> Good to know, thanks. Our component is designed to be 24px height and 14pt font for both the inline blank and dropdown choices. I think that will be better. I don't think we should go below 24px for the target area, and the font size is our standard. In which case the only implementation issue I see is the background color. Does all that sound good?

yeah, thanks. FYI @muHashh

**Comment by vabarbosa** [comment]
@muHashh thank you!

@qiskit-community/digital-applications-devs this is should be ready for review

@JRussellHuffman mind also taking a final pass?

you may preview examples of fill in the blanks in both block and inline here: https://platypus-pr-218.ecy0akwlcpw.us-south.codeengine.appdomain.cloud/course/ch-states/the-atoms-of-computation



- **exercise block**

    image

    image


- **inline**

    image

     image

    image

    image

**Comment by JRussellHuffman** [comment]
Looks great! The only issue I now see is with the shadow. It's a little heavy and looks like it sits in front of the blank above it. Take a look at this screenshot.

Screen Shot 2021-12-15 at 12 33 23 PM

File: frontend/scss/vendor/mathigon.scss [hunk 1]
```diff
@@ -4,6 +4,115 @@
 @import '../variables/mq.scss';
 @import '../variables/fonts.scss';
 
+x-blank, x-blank-mc {
+  .target {
+    background: $background-color-lighter;
```
Comment (techtolentino) [inline frontend/scss/vendor/mathigon.scss:9]: would this still work if we specified `background-color`?

**Review by techtolentino — CHANGES_REQUESTED** [review]
This looks great @vabarbosa - thank you... 

I just made a tiny suggestion regarding css properties... Also, I like Russell's suggestion about the `box-shadow`

I quickly messed with some of the css shadow values, and wanted to offer what I tested (via _inspecting elements_ - this might not be the solution, but just sharing... 🙏)

Screen Shot 2021-12-15 at 12 30 45 PM

File: frontend/scss/vendor/mathigon.scss (hunk 1 above)
Comment (vabarbosa) [reply frontend/scss/vendor/mathigon.scss]: yup, thanks... updates.

**Comment by vabarbosa** [comment]
@JRussellHuffman @techtolentino thanks.

the box shadow has been updated to correspond to the values used in the Figma design:

image

image


and updated to use `background-color` instead of `background`

**Review by techtolentino — APPROVED** [review]
thanks @vabarbosa - I think this looks good!

**Pull request merged**